### PR TITLE
added JsResultException as cause for RuntimeException

### PR DESCRIPTION
**Pull request by yarosman**
Wrap JsResultException with validations errors to Runtime Exception on JSONSerializationPack deserialize and readValue methods for simplify server error processing in HttpErrorHandler

### CODE REVIEW

File: src/main/scala/package.scala [hunk 1]
```diff
@@ -16,65 +16,11 @@
 package reactivemongo.play.json
 
 import scala.util.{ Failure, Success, Try }
-
```
Comment (cchantep) [inline src/main/scala/package.scala]: Please do not mix reformatting (there not compliant with project pref)

Comment (cchantep) [inline src/main/scala/package.scala]: This change still needs to be reverted

Comment (cchantep) [inline src/main/scala/package.scala]: This change still needs to be reverted

File: src/main/scala/package.scala [hunk 2]
```diff
@@ -554,12 +555,13 @@ object JSONSerializationPack extends reactivemongo.api.SerializationPack {
 
   def deserialize[A](document: Document, reader: Reader[A]): A =
     reader.reads(document) match {
-      case JsError(msg)    => sys.error(msg mkString ", ")
+      case JsError(msg)    => throw new RuntimeException(msg mkString ", ", JsResultException(msg))
```
Comment (cchantep) [inline src/main/scala/package.scala]: There `JsResultException` can be directly thrown

Comment (yarosman) [inline src/main/scala/package.scala]: Maybe `JsResultException` can be directly thrown in `writeToBuffer()` and `readValue()` methods too ?

**Pull request closed without merging**